### PR TITLE
[FIX] Cypht reply-all could have a better UX

### PR DESCRIPTION
**Pull request by jeanpaulHamuli**
Issue fixed : Cypht reply-all could have a better UX
Decription of issue : https://avan.tech/item57670-Cypht-reply-all-could-have-a-better-UX

### CODE REVIEW

File: modules/imap/output_modules.php [hunk 1]
```diff
@@ -243,7 +243,7 @@ protected function output() {
                     $txt .= ' | <a class="reply_all_link hlink" href="?page=compose&amp;reply_all=1'.$reply_args.'">'.$this->trans('Reply-all').'</a>';
                 }
                 else {
-                    $txt .= ' | <a class="reply_all_link hlink disabled_link">'.$this->trans('Reply-all').'</a>';
+                    $txt .= ' | <a class="reply_all_link hlink" href="?page=compose&amp;reply_all=1'.$reply_args.'">'.$this->trans('Reply-all').'</a>';
```
Comment (henrique-borba) [inline modules/imap/output_modules.php:246]: It looks good, have you checked if the code below can add the `disabled_link` class again without you wanting to?

https://github.com/jasonmunro/cypht/blob/9a61e208fbd462c02dc561bb2db6625ea4e7d353/modules/imap/site.js#L721-L724

**Review by henrique-borba — CHANGES_REQUESTED** [review]

File: modules/imap/output_modules.php (hunk 1 above)
Comment (jeanpaulHamuli) [reply modules/imap/output_modules.php:246]: It can only add the `disabled_link` when the page has not yet loaded

Comment (henrique-borba) [reply modules/imap/output_modules.php:246]: Good, I just realized that the `else` below is unnecessary:

https://github.com/jasonmunro/cypht/blob/8aadcbfe1cde41043c571af96992ba7f62670fd7/modules/imap/output_modules.php#L242-L247

The code is the same for both conditions

**Review by henrique-borba — CHANGES_REQUESTED** [review]

File: modules/imap/output_modules.php (hunk 1 above)
Comment (jeanpaulHamuli) [reply modules/imap/output_modules.php:246]: I didn't pay attention to that, let me remove one and the condition as it is no longer in use

File: modules/imap/output_modules.php [hunk 2]
```diff
@@ -239,12 +239,7 @@ protected function output() {
                 '<a class="hlink small_headers" style="display: none;" href="#">'.$this->trans('Small headers').'</a>';
             if (!isset($headers['Flags']) || !stristr($headers['Flags'], 'draft')) {
                 $txt .= ' | <a class="reply_link hlink" href="?page=compose&amp;reply=1'.$reply_args.'">'.$this->trans('Reply').'</a>';
-                if ($reply_all || $size > 1) {
-                    $txt .= ' | <a class="reply_all_link hlink" href="?page=compose&amp;reply_all=1'.$reply_args.'">'.$this->trans('Reply-all').'</a>';
-                }
-                else {
-                    $txt .= ' | <a class="reply_all_link hlink disabled_link">'.$this->trans('Reply-all').'</a>';
-                }
+                $txt .= ' | <a class="reply_all_link hlink" href="?page=compose&amp;reply_all=1'.$reply_args.'">'.$this->trans('Reply-all').'</a>';
```
Comment (henrique-borba) [inline modules/imap/output_modules.php:242]: We still need to check if there is more than one sender to reply

```php
if ($reply_all || $size > 1) {
   $txt .= ' | <a class="reply_all_link hlink" href="?page=compose&amp;reply_all=1'.$reply_args.'">'.$this->trans('Reply-all').'</a>';
}
```

Comment (jeanpaulHamuli) [reply modules/imap/output_modules.php:242]: If we still check without `else`, the **reply all** button will disappear when there is less than one message. 

It should be like that ?

Comment (henrique-borba) [reply modules/imap/output_modules.php:242]: You're right, the previous pattern was to show so let's keep it that way. Thanks Jean

Comment (henrique-borba) [reply modules/imap/output_modules.php:242]: You're right, the previous behaviour was to show it anyway so let's keep it that way. Thanks Jean

**Review by henrique-borba — APPROVED** [review]

File: modules/imap/output_modules.php (hunk 2 above)
Comment (jeanpaulHamuli) [reply modules/imap/output_modules.php:242]: Thank you too Henrique

**Pull request merged**